src/client/Client.cc

```diff
@@ -3114,12 +3114,24 @@ void Client::handle_mds_map(const MConstRef<MMDSMap>& m)
 	continue;
       }
       if (newstate >= MDSMap::STATE_ACTIVE) {
-	if (oldstate < MDSMap::STATE_ACTIVE) {
-	  // kick new requests
-	  kick_requests(session.get());
-	  kick_flushing_caps(session.get());
-	  signal_context_list(session->waiting_for_open);
-	  wake_up_session_caps(session.get(), true);
+	if (oldstate <= MDSMap::STATE_ACTIVE && newstate != oldstate) {
```

Comment:
this doesn't make sure that the MDS stays in the stopping state while the dirty caps are being flushed, is there any way we can lock the state of MDS for the time being so that the cap flush is complete?

Reply:
No. The stopping MDS may export the caps eventually so the client will reconnect the caps to the destination (rank 0).

Reply:
so you mean the stopping MDS would provide the caps to the client it needs; goes to the new destination (rank 0) where it reconnects with the same set of caps, and there is no replay of the client requests, right?

Reply:
The request might wait on a frozen queue but eventually get forwarded to the auth mds. (stopping mds drain to rank 0)